pkg/auth/oauth/external/gitlab/gitlab.go

```diff
@@ -0,0 +1,111 @@
+package gitlab
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/RangelReale/osincli"
+	"github.com/golang/glog"
+
+	authapi "github.com/openshift/origin/pkg/auth/api"
+	"github.com/openshift/origin/pkg/auth/oauth/external"
+)
+
+const (
+	gitlabAuthorizePath = "oauth/authorize"
+	gitlabTokenPath     = "oauth/token"
+	gitlabUserApiPath   = "api/v3/user"
```

Comment:
The possibility to use GitLab as an OAuth provider was introduced with version 7.7.0 (mid of 2014) when the API version already was on v3. I added a note in code which version is required (I will also update openshift/openshift-docs#1503 accordingly).